spock-core/src/main/java/org/spockframework/runtime/extension/builtin/RetryIterationInterceptor.java

```diff
@@ -40,11 +46,10 @@ public RetryIterationInterceptor(Retry retry, MethodInfo featureMethod) {
 
   @Override
   public void intercept(IMethodInvocation invocation) throws Throwable {
-    iterationState.startIteration();
     for (int i = 0; i <= retry.count(); i++) {
-      iterationState.setRetryAttempt(i);
+      iterationState.setRetryAttempt(invocation.getIteration(), i);
```

Comment:
Can we rewrite this now that we have a proper store implementation?

Reply:
Definitely :-)